crates/bevy_render/src/extract_instances.rs

```diff
@@ -27,64 +27,59 @@ use crate::{prelude::ViewVisibility, Extract, ExtractSchedule, RenderApp};
 /// This is essentially the same as
 /// [`ExtractComponent`](crate::extract_component::ExtractComponent), but
 /// higher-performance because it avoids the ECS overhead.
-pub trait RenderInstance: Send + Sync + Sized + 'static {
+pub trait ExtractInstance: Send + Sync + Sized + 'static {
     /// ECS [`WorldQuery`] to fetch the components to extract.
     type Query: WorldQuery + ReadOnlyWorldQuery;
     /// Filters the entities with additional constraints.
     type Filter: WorldQuery + ReadOnlyWorldQuery;
 
     /// Defines how the component is transferred into the "render world".
-    fn extract_to_render_instance(item: QueryItem<'_, Self::Query>) -> Option<Self>;
+    fn extract(item: QueryItem<'_, Self::Query>) -> Option<Self>;
 }
 
 /// This plugin extracts one or more components into the "render world" as
-/// render instances.
+/// extracted instances.
 ///
 /// Therefore it sets up the [`ExtractSchedule`] step for the specified
-/// [`RenderInstances`].
+/// [`ExtractedInstances`].
 #[derive(Default)]
-pub struct RenderInstancePlugin<RI>
+pub struct ExtractInstancesPlugin<EI>
 where
-    RI: RenderInstance,
+    EI: ExtractInstance,
 {
     only_extract_visible: bool,
-    marker: PhantomData<fn() -> RI>,
+    marker: PhantomData<fn() -> EI>,
 }
 
-/// Stores all render instances of a type in the render world.
+/// Stores all extract instances of a type in the render world.
 #[derive(Resource, Deref, DerefMut)]
-pub struct RenderInstances<RI>(EntityHashMap<Entity, RI>)
+pub struct ExtractedInstances<EI>(EntityHashMap<Entity, EI>)
 where
-    RI: RenderInstance;
+    EI: ExtractInstance;
 
-impl<RI> Default for RenderInstances<RI>
+impl<EI> Default for ExtractedInstances<EI>
 where
-    RI: RenderInstance,
+    EI: ExtractInstance,
 {
     fn default() -> Self {
         Self(Default::default())
     }
 }
 
-impl<RI> RenderInstancePlugin<RI>
+impl<EI> ExtractInstancesPlugin<EI>
 where
-    RI: RenderInstance,
+    EI: ExtractInstance,
 {
-    /// Creates a new [`RenderInstancePlugin`] that unconditionally extracts to
+    /// Creates a new [`ExtractInstancesPlugin`] that unconditionally extracts to
     /// the render world, whether the entity is visible or not.
     pub fn new() -> Self {
         Self {
             only_extract_visible: false,
             marker: PhantomData,
         }
     }
-}
 
-impl<RI> RenderInstancePlugin<RI>
-where
-    RI: RenderInstance,
-{
```

Comment:
@pcwalton / @superdump  I don't know whether there is something behind I'm missing (I'm quite a new in Rust). There is defined twice:

```
impl<EI> ExtractInstancesPlugin<EI>
where
    EI: ExtractInstance,
{
```

So it can be merged as one definition with two methods.

Is that correct?

Reply:
As far as I understand, yes. You can have multiple impl blocks but in this case it looks like they could be merged together.